write.js

```diff
@@ -355,12 +356,19 @@ export function addWriteMethods(
 
 		return (callback) => {
 			if (writtenBatchDepth) {
-				// if we are in a batch, the transaction can't close, so we do the faster,
+				// If we are in a batch, the transaction can't close, so we do the faster,
 				// but non-deterministic updates, knowing that the write thread can
-				// just poll for the status change if we miss a status update
-				writeStatus = uint32[flagPosition];
-				uint32[flagPosition] = flags;
-				//writeStatus = Atomics.or(uint32, flagPosition, flags)
+				// just poll for the status change if we miss a status update.
+				// That is, if we are on x64 architecture...
+				if (arch === 'x64') {
+					writeStatus = uint32[flagPosition];
+					uint32[flagPosition] = flags;
```

Comment:
To be consistent with `Atomics.or` this should be a `|=`, but if `uint32` is made up of only `0` s then it makes no difference.